services/address_book_service.proto

```diff
@@ -29,37 +27,127 @@ import "transaction_response.proto";
+ *   software SHALL
+ *    - Validate the threshold signature for the Hedera governing council
+ *    - Validate the signature of the "admin key" for the account assigned
+ *      as the "node account", _if present_.
+ *    - Modify the existing node to a pending delete state.
+ *    - When executing the next `freeze` transaction with `freeze_type` set to
+ *      `PREPARE_UPGRADE`, update network configuration and transition the
+ *      node pending deletion to a "deleted" state.
+ *
+ * ### For a node update transaction.
+ * - The node operator or Hedera council representative SHALL create an
+ *   `updateNode` transaction.
+ *    - If the node operator creates the transaction
+ *       - The node operator MUST sign this transaction with the "admin key" of
+ *         the account assigned as the "node account".
+ *       - If the transaction changes the value of the "node account" the
+ *         node operator MUST _also_ sign this transaction with the "admin key"
+ *         of the account to be assigned as the new "node account".
+ *       - The node operator SHALL submit the transaction to the
+ *         network.  Hedera council approval SHALL NOT be sought for this
+ *         transaction
+ *    - If the Hedera council representative creates the transaction
+ *       - The Hedera council representative SHALL arrange for council members
+ *         to review and sign the transaction.
+ *       - Once sufficient council members have signed the transaction, the
+ *         Hedera council representative SHALL submit the transaction to the
+ *         network.
+ * - Upon receipt of a valid and signed node update transaction the network
+ *   software SHALL
+ *    - If the transaction is signed by the Hedera governing council
+ *       - Validate the threshold signature for the Hedera governing council
+ *    - If the transaction is signed by the node account "admin key"
+ *       - Validate the signature of the "admin key" for the account assigned
+ *         as the "node account".
+ *    - Create a "pending modification" in network state with the changes
+ *      requested in the update transaction.
+ *    - When executing the next `freeze` transaction with `freeze_type` set to
+ *      `PREPARE_UPGRADE`, update network configuration according to the
+ *      "pending modification" and merge the pending state to active state.
  */
 service AddressBookService {
     /**
-      * Prepare to add a new node to the network.
-      * When a valid council member initiates a HAPI transaction to add a new node,
-      * then the network should acknowledge the transaction and update the network’s Address Book within 24 hours.
-      * The added node will not be active until the network is upgraded.
-      */
-    rpc createNode (Transaction) returns (TransactionResponse);
+     * A transaction to create a new consensus node in the network.
+     * address book.<br/>
+     * This transaction, once complete, SHALL add a new consensus node to the
```

Comment:
Even this transaction is complete, the new node is still not joining the network until next freeze upgrade for phase 2 and not joining the network until consensus roster sent in the next few round for phase 3 (more details still pending to be narrowed down.)

Reply:
I'll clarify that the node is added to _state_ (which it must be, or the consensus roster will never be updated in a later process).